### DISCUSSION TITLE
Does this work for react native?

### DISCUSSION THREAD

**Discussion by Cmoen11** [kaciakmaciak/react-query-subscription#49] [Q&A] [answered]
Hi. I'm trying to implement this in my React Native project, and I'm getting stuck by it not finding "dist/react-query-subscription.umd.js",

I saw in package.json that it had a reference to "browser": "dist/react-query-subscription.umd.js",

I removed this and the issue went away... Now I'm facing 

```
{"isTrusted": false, "message": "Invalid Sec-WebSocket-Accept response"}
```

and I'm unsure if this is related or not.

Any ideas?

**Comment by kaciakmaciak** [comment]
> Hi. I'm trying to implement this in my React Native project, and I'm getting stuck by it not finding "dist/react-query-subscription.umd.js",
> 
> 
> 
> I saw in package.json that it had a reference to "browser": "dist/react-query-subscription.umd.js",
> 
> 
> 
> I removed this and the issue went away... Now I'm facing 
> 
> 
> 
> ```
> 
> {"isTrusted": false, "message": "Invalid Sec-WebSocket-Accept response"}
> 
> ```
> 
> 
> 
> and I'm unsure if this is related or not.
> 
> 
> 
> Any ideas?
> 
> 

Hi Christian,

I haven't had much experience with react native but I'm definitely keen on making it work. I'll have a look over the weekend and will try to resolve the issue.

**Comment by kaciakmaciak** [comment, marked as answer]
Hi Christian,

Thanks for reporting the issue with the `browser` key in the `package.json`. It's been fixed now and released [v1.5.1](https://github.com/kaciakmaciak/react-query-subscription/releases/tag/v1.5.1) 🙏 

Regarding your question about the "Invalid Sec-WebSocket-Accept response" error. I've created a react native project using expo and installed `react-query-subscritpion` (alongside of its peer dependencies) and had no problems. I've created a very basic app, you can see it here https://github.com/kaciakmaciak/react-native-react-query-subscription-sandbox. I've also tried using websockets over localhost (the connection URL had to be something like `ws://192.168.x.x:port/path` rather than just using `ws://localhost:port/path`). It worked all good once I had a correct URL.

My gut feeling is that your error "Invalid Sec-WebSocket-Accept response" is related to WebSocket connection itself rather than `react-query-subscritpion`. You could verify this by connecting to your websocket directly, without using `useSubscription`.

Let us know if have any more questions.

**Reply by Cmoen11** [reply]
Thanks @kaciakmaciak 

I appreciate how fast this was resolved and I'd debug my issues :)

**Reply by kaciakmaciak** [reply]
No worries, let us know if any problems.
